### DISCUSSION TITLE
Add shops and restaurants to the "Check if element still exist" quest?

### DISCUSSION THREAD

**Discussion by stoofvlees** [streetcomplete/StreetComplete#4232] [Q&A] [answered]
Depending on the area, some shops and restaurants tend to change owners pretty often. I think it could be useful to add shops and restaurants to the "Check if element still exist" quest, so that it's easy to delete or update them and keep the map up to date.
Any thoughts?

**Comment by JLZIMMERMANN** [comment]
In my practice I use survey:date and when there is a change I keep the old tags using
was:name=*
was:shop=* and so and so

**Reply by stoofvlees** [reply]
> 

Indeed valuable to keep some history. That's why I usually keep the same node, but only change the tags, so that the node history is kept and can be viewed by anyone. Using "was:" is another way to achieve this goal.

**Comment by matkoniecz** [comment]
In practice opening hours and other quests act as de facto existence checker. Have you disabled them?

**Reply by stoofvlees** [reply]
> In practice opening hours and other quests act as de facto existence checker. Have you disabled them?

Thanks for pointing this out. Indeed I disabled the opening hours quest. For me, adding opening hours during a walk through town is just too finnicky and has little value over adding opening hours by armchair mapping using the shop's website as source. So, to me, it's debatable whether the opening hours quest meets the "Easy answer" and "Valuable Surveyors" considerations/guidelines.
Also, I wonder if the opening hours question is repeated from time to time, like the existence quest?

**Reply by matkoniecz** [reply]
> I wonder if the opening hours question is repeated from time to time, like the existence quest?

Yes, and it has a special mode which allows you to avoid reentering things from scratch.

> over adding opening hours by armchair mapping using the shop's website as source

I guess that depends on availability and quality of shop websites (in my area websites tend to be outdated or nonexisting while copying from Google Maps violates OSM rules)

> Indeed I disabled the opening hours quest.

In such case it qualifies for opening an issue, I guess. See https://github.com/streetcomplete/StreetComplete/issues

I will keep it on my list of ideas to test, but making issue makes less likely that it will be lost.

**Reply by peternewman** [reply]
Likewise I have it disabled as it's very fiddly to complete initially and for brands at least, I'd trust their website more than OSM as they change a lot or through the year for things like holidays.

See also #4231 which sort of overlaps, especially my comment https://github.com/streetcomplete/StreetComplete/discussions/4231#discussioncomment-3256300

**Comment by matkoniecz** [comment]
On trying to implement it and thinking about it: note that for shops which are typically replaced by other shops it is preferable to have this special shop replacement menu that allows to select a new POI present there. So this would require a brand new quests, not just adding it to existence checker.

**Reply by westnordost** [reply]
Not necessarily. You could add logic to CheckExistenceForm for example

**Reply by matkoniecz** [reply]
There is one other reason: check existence is before opening hours.

This one should be after opening hours, otherwise they would not be resurveyed in some cases. For example if check existence AND opening hours quests are enabled AND shop has `opening_hours` tagged AND shop node was not edited for 13 years, then opening hours resurvey quest should be asked, not check existence quest.

Otherwise user would confirm existence, what would update object and opening hours resurvey would not be asked.

Separate quest late in the list would also allow to use some shorter recheck time.

**Comment by matkoniecz** [comment, marked as answer]
implemented in https://github.com/streetcomplete/StreetComplete/pull/4843

@stoofvlees thanks for the idea! I would never invent it as I edit with opening hours on.

I will convert it to QA and mark answer to mark it as done on the discussion overview.

**Reply by stoofvlees** [reply]
That's truly great! I think it's a valuable addition that can help keeping things up to date. Keep up the good work! :-)

**Comment by dbdean** [comment]
@matkoniecz, this new quests looks good, but I wonder why it is only looking at nodes and not also ways/areas like the opening hours quest is?

It would also be good to include a few other things like hotels, etc that never get asked about opening hours, but would be very useful to check on their existence every so often.

**Reply by matkoniecz** [reply]
> and not also ways/areas like the opening hours quest is?

SC can delete nodes, cannot delete areas and ways. OH (I think) asks user to leave note in case of selecting that it does not exist for ways/areas 

(from memory, not tested/checked)

**Reply by dbdean** [reply]
In the OH quest, you can indicate that it is now vacant or changed - which is generally the case. In the case that it the item is gone and can't be made vacant, then yes - you would leave a note.

There are plenty of shops that are mapped as areas, and ignoring them here seems to be a pity.

**Reply by westnordost** [reply]
I agree that it would be useful if the existence of shops as areas was asked too. However, it would need to be tested what happens when one selects "does not exist anymore". IIRC @matkoniecz is wrong, the element is not deleted but re-tagged as vacant in that case, so it doesn't matter whether it is a node or way. Feel free to create a ticket for that or a PR

**Reply by matkoniecz** [reply]
yes, I was wrong, and actually nodes are not able to be deleted

changed in https://github.com/streetcomplete/StreetComplete/commit/a3569b1e8ab787ae5a3e36c1d476b691fc2bb2ea

thanks for reporting this!
